src/routes/index.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { BrowserRouter as Router, Switch } from 'react-router-dom';
+import HomePage from '../pages/HomePage';
+import PublicRoute from './PublicRoute';
+import CategoryPage from '../pages/CategoryPage';
+import { useSelector, useDispatch } from "react-redux";
+import { useState, useEffect } from 'react';
+import { setToken } from '../actions';
+import { getTopicsData } from '../actions/topicsDataSetter';
+import SearchResult from '../components/SearchResult';
+import { getToken } from '../actions';
+import PrivateRoute from './PrivateRoute';
+import ProfilePage from '../components/ProfilePage';
+import { getUserProfile } from '../actions/profile';
+
+
+const Routes = props => {
+    const [mounted, setMounted] = useState(false)
+
+    const dispatch = useDispatch();
+
+    const myState = useSelector((state) => state.authReducer);
+
+    const profile = useSelector(state=> state.profile);
+    const {userProfile} = profile;
+
+    if (!mounted) {
+        dispatch(getTopicsData());
+    }
+
+
+    useEffect(() => {
+        setMounted(true);
+    }, []);
```

Comment:
I am confused with this. Why are we not just calling the `getTopicsData` action? Also we can use `useEffect` to achieve required lifecycle behaviour, Go through this [stackoverflow answer](https://stackoverflow.com/a/54655508)